src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSApiMethodParamTransformer.java

```diff
@@ -14,24 +14,165 @@
+    paramDoc.paramName(context.getNamer().localVarName(Name.from("request")));
+    paramDoc.lines(ImmutableList.of("The request object that will be sent."));
+
+    String typeName = "Object";
+    Iterable<Field> optionalParams = removePageTokenFromFields(methodConfig);
+    if (!methodConfig.getRequiredFieldConfigs().iterator().hasNext()
+        && !optionalParams.iterator().hasNext()) {
+      typeName += "=";
+    }
+
+    paramDoc.typeName(typeName);
+    return paramDoc.build();
+  }
+
+  private ParamDocView generateOptionsParamDoc() {
+    SimpleParamDocView.Builder paramDoc = SimpleParamDocView.newBuilder();
+    paramDoc.paramName("options");
+    paramDoc.typeName("Object=");
+    paramDoc.lines(
+        ImmutableList.of(
+            "Optional parameters. You can override the default settings for this call, e.g, timeout,",
+            "retries, paginations, etc. See [gax.CallOptions]{@link "
+                + "https://googleapis.github.io/gax-nodejs/global.html#CallOptions} for the details."));
+    return paramDoc.build();
+  }
+
+  private List<Field> removePageTokenFromFields(MethodConfig methodConfig) {
+    ImmutableList.Builder<Field> newFields = ImmutableList.builder();
+    for (Field field : methodConfig.getOptionalFields()) {
+      if (methodConfig.isPageStreaming()
+          && field.equals(methodConfig.getPageStreaming().getRequestTokenField())) {
+        continue;
+      }
+      newFields.add(field);
+    }
+    return newFields.build();
+  }
+
+  private List<ParamDocView> generateMethodParamDocs(
+      MethodTransformerContext context, Iterable<Field> fields, boolean isOptional) {
+    SurfaceNamer namer = context.getNamer();
+    MethodConfig methodConfig = context.getMethodConfig();
+    ImmutableList.Builder<ParamDocView> docs = ImmutableList.builder();
+    for (Field field : fields) {
+      if (isRequestTokenParam(methodConfig, field)) {
+        continue;
+      }
+
+      SimpleParamDocView.Builder paramDoc = SimpleParamDocView.newBuilder();
+      paramDoc.paramName("request." + namer.getVariableName(field));
+
+      String typeName = namer.getFieldTypeDoc(context.getTypeTable(), field);
+      paramDoc.typeName(typeName + (isOptional ? "=" : ""));
+
+      ImmutableList.Builder<String> docLines = ImmutableList.builder();
+      if (isPageSizeParam(methodConfig, field)) {
+        docLines.add(
+            "The maximum number of resources contained in the underlying API",
+            "response. If page streaming is performed per-resource, this",
+            "parameter does not affect the return value. If page streaming is",
+            "performed per-page, this determines the maximum number of",
+            "resources in a page.");
+      } else {
+        docLines.addAll(namer.getDocLines(field));
+      }
+
+      if (field.getType().isMessage() && !field.getType().isMap()) {
+        // Add an empty line if there was doclines added previously
+        if (docLines.build().size() > 0) {
```

Comment:
This line looks like applying immutable pattern wrongly; I believe ImmutableList should be built only once. You should have a boolean flag for this (which should be configured around line 145 based on whether `namer.getDocLines(field)` is empty or not), or you can just use an `ArrayList`.